common/include/pcl/common/impl/centroid.hpp

```diff
@@ -200,15 +206,15 @@ computeCovarianceMatrix (const pcl::PointCloud<PointT> &cloud,
       pt[1] = point.y - centroid[1];
       pt[2] = point.z - centroid[2];
 
-      covariance_matrix (1, 1) += pt.y () * pt.y ();
-      covariance_matrix (1, 2) += pt.y () * pt.z ();
+      accu (1, 1) += pt.y () * pt.y ();
+      accu (1, 2) += pt.y () * pt.z ();
 
-      covariance_matrix (2, 2) += pt.z () * pt.z ();
+      accu (2, 2) += pt.z () * pt.z ();
 
       pt *= pt.x ();
-      covariance_matrix (0, 0) += pt.x ();
-      covariance_matrix (0, 1) += pt.y ();
-      covariance_matrix (0, 2) += pt.z ();
+      accu (0, 0) += pt.x ();
+      accu (0, 1) += pt.y ();
+      accu (0, 2) += pt.z ();
```

Comment:
This one would also be eligible for a full block operation. The covariance can be computed in  a single matrix multiplication.